pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -160,51 +188,132 @@ const CryptoPolicyDialog = ({
+                <ShutdownModal onClose={() => setShowShutDownModal(false)} />
+                }
+            </li>
+        );
     }
 
+    if (!requiresReboot && currentCryptoPolicy === "FIPS" && !fipsEnabled) {
+        return (
+            <li className="system-health-crypto-policies">
+                <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }}>
+                    <FlexItem><ExclamationTriangleIcon size="sm" className="crypto-policies-health-card-icon" /></FlexItem>
+                    <div>
+                        <div id="fips_mode_disabled_text">
+                            {_("FIPS crypto policy selected, but not active")}
+                        </div>
+                        <Button isInline variant="link" className="pf-u-font-size-sm" onClick={() => setShowEnableFIPSModal(true)}>
+                            {_("Enable FIPS mode and reboot")}
+                        </Button>
+                    </div>
+                </Flex>
+                {showEnableFIPSModal &&
+                <EnableFIPSModal onClose={() => setShowEnableFIPSModal(false)} />
+                }
+            </li>
+        );
+    }
+
+    return null;
+};
+
+const EnableFIPSModal = ({ onClose }) => {
+    const [error, setError] = useState();
+    const [inProgress, setInProgress] = useState(undefined);
+
+    const onSubmit = () => {
+        setInProgress(true);
+        cockpit.spawn(["fips-mode-setup", "--enable"], { superuser: "require", err: "message" })
+                .then(() => {
+                    cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
+                            .catch(error => { setInProgress(false); setError(error) });
+                })
+                .catch(error => { setInProgress(false); setError(error) });
+    };
+
     return (
-        <li className="system-health-crypto-policies">
-            <Flex flexWrap={{ default: 'nowrap' }} spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }}>
-                <InProgressIcon size="sm" className="crypto-policies-health-card-icon" />
-                <Button isInline variant="link" onClick={() => setShowShutDownModal(true)}>
-                    {_("Reboot to apply new crypto policy")}
-                </Button>
-            </Flex>
-            {showShutdownModal &&
-            <ShutdownModal onClose={() => setShowShutDownModal(false)} />
-            }
-        </li>
+        <Modal isOpen position="top" variant="medium"
+                   onClose={onClose}
+                   id="enable-fips-mode-dialog"
+                   title={_("Enable FIPS mode")}
+                   footer={<>
+                       {error && <ModalError dialogError={typeof error == 'string' ? error : error.message} />}
+                       <Button variant='primary' isDisabled={inProgress} isLoading={inProgress} onClick={onSubmit}>{_("Enable FIPS mode and reboot")}</Button>
```

Comment:
Could we do a similar deferral of rebooting as in the crypto policy dialog? The time when spotting this error is not necessarily a good time to reboot your machine.

Reply:
Right, but that's exactly the same as with changing the crypto policy. It's oddly inconsistent, and rather than forcing a reboot now, it will more likely lead to ignoring the issue. We wouldn't *have* to have two health card statuses -- both kinds of reboot (after update-crypto-policy and after fips-mode-setup) are the same kind of root cause, so there should be just one.

Reply:
I think the difference is that with FIPS mode you really have to mandatory reboot for it to work, while other policies can work somewhat without rebooting. It is indeed inconsistent but for crypto-policies we don't handle this case (when an admin changes a crypto policy without a reboot we can't detect it afaik.

Reply:
So are you saying we should remove the inconsistent defer for the normal one too? 

Sure; I guess that would mean less code, less UI, and fewer things to go wrong for our users.

The inconsistent state would only be for when someone changes things outside of Cockpit then?

Reply:
Conclusion from our meeting:

* Change the crypto policies dialog to always reboot
* Switching away from FIPS crypto policy disables FIPS mode

Inconsistent state can happen with FIPS mode:

* Admin changes to FIPS crypto policy with enabling fips mode
* Admin has fips mode enabled but uses a default policy
* Admin enables fips mode setup but didn't reboot

In this case we show a healthcard entry and let the user re-apply the current crypto policy and reboot so it's in a consistent state again.